buildenv/jenkins/common/build.groovy

```diff
@@ -864,10 +864,13 @@ def recompile() {
             LOG_LINES.addAll(newLines)
 
             if (!match_fail_pattern(newLines)) {
-                //different error
+                // different error
                 archive_diagnostics()
                 throw f
             }
+            if (retryCounter >= maxRetry) {
+                throw f
+            }
```

Comment:
Suggest this should also call `archive_diagnostics()` and can occur before checking the build log.

Reply:
I thought about doing the `archive_diagnostics()`, but if it's a known issue (matches the pattern) I think it would just be a waste of time and space.

Reply:
I hadn't thought about it that way: I agree.